gmi_egads/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+if(DEFINED TRIBITS_PACKAGE)
+  include(pkg_tribits.cmake)
+  return()
+endif()
+
+if(NOT ENABLE_EGADS)
+  return()
+endif()
+
+option(EGADS_LITE "Enable EGADSlite" OFF)
+# this file brings EGADS_LITE from CMake to C++
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/gmi_egads_config.h.in"
+               "${CMAKE_CURRENT_BINARY_DIR}/gmi_egads_config.h")
+
+#Sources & Headers
+set(SOURCES gmi_egads.c)
+set(HEADERS gmi_egads.h)
+
+add_library(gmi_egads ${SOURCES})
+
+target_include_directories(gmi_egads PUBLIC ${EGADS_INCLUDE_DIR})
```

Comment:
Cursory look over indicates that egads isn't part of the interface, so you should be using PRIVATE linkage and header file includes

Reply:
I've got no objections to switching to private linkage, but I want to make sure that some of the symbols are not internal. While they're not part of the public interface for PUMI (and thus not in the header file), I use `struct gmi_model* gmi_egads_init(struct egObject *body, int numRegions)` and `void gmi_egads_init_adjacency(int ***adjacency)` in the code that generates a mesh in the PUMI format from an EGADS model, so I need those functions to be callable from an external project. I don't think I know enough about cmake to make that happen, or if that is even what you're really referring to.

Reply:
The `EGADS_INCLUDE_DIR` and `EGADS_LIBRARIES` point to the external egads libraries. If you are doing a public linkage/include you are requiring that the user of the compiled pumi library can also link against those libs. If you make the linkage private then you will need to have egads to compile pumi with egads support, but whoever consumes pumi will not need to have access to egads to compile/link. (although you will need the egads library at runtime if you do dynamic linkage)

Your functions `gmi_egads_init` and `gmi_egads_init_adjacency` are part of the `gmi_egads` library target that you create, and you want those visible to anyone who consumes your library, so adding the `CMAKE_CURRENT_SOURCE_DIR` and `include` as public targets is correct.

Reply:
Okay, I think I've addressed this in ad8ab1f.

Reply:
Looks good 😃